src/java/voldemort/store/readonly/swapper/AdminStoreSwapper.java

```diff
@@ -238,9 +243,29 @@ private String fetch(String storeDir) {
         }
 
         // Invalid stores should fail faster
-        if(invalidBootstrapURLExceptions) {
-            throw new InvalidBootstrapURLException("Exceptions during push. Invalid bootstrap url. Please check your "
+        /*if(invalidBootstrapURLExceptions) {
+           throw new InvalidBootstrapURLException("Exceptions during push. Invalid bootstrap url. Please check your "
                                                    + "cluster bootstrap URL");
+
+        }*/
+        ArrayList<Node>failedNodes = new ArrayList<Node>();
+        for (final Node node: cluster.getNodes()){
+            Future<String> val = fetchDirs.get(node.getId());
+            try{
+                String response = val.get();
+                fetchResponseMap.put(node, new Response(response));
+            }catch(Exception e){
+                if (e.getCause() instanceof InvalidBootstrapURLException){
+                    throw new InvalidBootstrapURLException("Not able to find store (" + storeName +
+                            ") in this cluster according to the push URL. BnP job is not able to create new stores now." +
+                            "Please create new stores through Nuage if you think this is the correct cluster you want to push." );
```

Comment:
Hmm... let's not mention Nuage in our error messages, since that system is not open-source (at least not yet) so it doesn't belong here.

Also, can you include ```e``` as a nested exception? This makes it easier to understand the stacktrace and root cause of issues.